### ISSUE TITLE
Wrong author when linking from one forum thread to another.

### ISSUE THREAD

**Issue by Tricky1975** [gamejolt/issue-tracker#640]
Darkblood tried to explain it here, but let's go into the deep of this:
http://gamejolt.com/f/forum-linking-bug/15476?page=0#forum-post-78426

Say you are in a thread of mine. If in that thread is a link to a thread by Jani and you click it, you will see Jani's thread, even with the subject and text Jani typed, but the system will claim the thread is mine, showing my name + avatar. If you refresh the page, my name and avatar disappear and get replaced by Jani's accordingly.

Second.... If you link from my thread to Jani's and you reply to Jani's thread, the reply ends up in my thread in stead of Jani's. I think this is caused by the same bug.

**Comment by Tricky1975** [comment]
Addition: I also found out, the "channel" the thread belongs to is not right when cross-linking between threads. I deem this likely caused by the same bug.

**Comment by Tricky1975** [comment]
I've tested something more in [this thread](http://gamejolt.com/f/test/70557) (Since I'm a moderator now I could). 

As you can see that thread contains nothing more than a link to Honno's "Sandbox" thread, and this test thread itself is closed. I clicked the Link to Honno's Sandbox thread (which the system will then show to be mine), and I placed a reply. The system refuses it, as... indeed it thinks it's still in my original thread, and that thread is closed, so that reply could not be added there. Since Honno's sandbox thread is still open the reply box did appear,but any replies will be rejected unless I refresh the page.

I did perform this test in order to find out how things would behave with this bug if closed threads were involved, well, this is the answer ;)

**Comment by Tricky1975** [comment]
I also found out this even affects the moderation tools (Sorry for telling that in public, but this is not that harmful I think). I once clicked through somebody else's thread to the "FAQ" thread on gj-help. Since that thread is closed I need to use the moderation tools to edit it, but, when I clicked the edit feature I did get to edit the thread I linked from, and not the thread I was supposed to edit.
